TaskManagerWebApi/Controllers/AccountControler.cs

```
@@ -60,6 +61,30 @@ public async Task<IActionResult> CreateUser([FromBody] UserAddRequest userAddReq
             return Ok(account.Value);
         }
 
+        [HttpPut("{userId}")]
+        [Authorize(AuthorizationPoilicyConstants.ADMIN_POLICY)]
+
+        public async Task<IActionResult> UpdateUser(int userId, [FromBody] UserUpdateRequest updatedUserRequest)
+        {
+            try
+            {
+                
+                var userEntity = mapper.Map<User>(updatedUserRequest);
+
+                var result = await accountService.UpdateUser(userId, userEntity, updatedUserRequest.NewPassword);
+
+                if (!result.IsSuccess)
+                {
+                    return BadRequest(result.Errors);
+                }
+
+                return Ok("User updated successfully");
+            }
+            catch (Exception ex)
+            {
+                return BadRequest(ex.Message);
+            }
```

why are you throwing and catching the exceptions?
Since you are using Fluent.Results library, those are wrapped automaticly in the Result<T>.
In case if you need to return the error (for instance, user not found) you have to retun Result.Fail(new Error())

Please remove all exceptuion catches, besides those that are relevant (only if you are using 3rd party library that might throw them)